lib/shopping_cart.rb

```diff
@@ -0,0 +1,45 @@
+class ShoppingCart
+
+  attr_reader :name, :capacity, :products, :details
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.chomp("items").to_i
+    @products = []
+    @details_hash = {}
+    @sum = 0
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    @details_hash[:name] = @name
+    @details_hash[:capacity] = @capacity
+    @details_hash
+  end
+
+  def total_number_of_products
+    products.each do |product|
+      @sum += product.quantity
+    end
+    @sum
```

Comment:
similarly here, I would take the sum out of the initialize method because in order for it to be populated this method needs to be called. The sum of the products might be more than 0, but if you try calling `sum` before you call this method, you'll still get 0.